### PR TITLE
Browser web classes

### PR DESCRIPTION
**Pull request by nshiab**
To load data into the SimpleData, I created three new methods:
- loadDataFromArray
- loadDataFromLocalFile
- loadDataFromUrl (which is async)

Because one of them is an async function, I also needed an async decorator: asynLogCall.

At first, we thought to load the data in the constructor, but it's impossible to deal with async call inside it.

So I went back to the method function.

Next step: create two classes of SimpleData: one for the web and one for NodeJS.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,13 +78,13 @@
...
         "@mui/material": "^5.8.0",
         "@observablehq/plot": "^0.4.3",
-        "@types/d3-scale": "^4.0.2",
         "@types/react": "^18.0.9",
         "@types/react-dom": "^18.0.4",
         "axios": "^0.27.2",
         "d3-array": "^3.1.6",
         "d3-scale": "^4.0.2",
         "d3-time-format": "^4.1.0",
+        "i": "^0.3.7",
```
[the remark this answers is not in the document]
Comment (nshiab) [reply package.json:87]: Good catch! It was a mistake I think!

File: src/class/SimpleData.ts [hunk 2]
```diff
@@ -30,14 +30,13 @@ import addItems_ from "../methods/addItems.js"
 import getUniqueValues_ from "../methods/getUniqueValues.js"
 import summarize_ from "../methods/summarize.js"
 import mergeItems_ from "../methods/mergeItems.js"
-import saveData_ from "../methods/saveData.js"
-import saveChart_ from "../methods/saveChart.js"
-import saveCustomChart_ from "../methods/saveCustomChart.js"
 import checkKeys from "../helpers/checkKeys.js"
 import logCall from "../helpers/logCall.js"
+import asyncLogCall from "../helpers/asyncLogCall.js"
```
[the remark this answers is not in the document]
Comment (nshiab) [reply src/class/SimpleData.ts:35]: Good idea. It's a quick fix. Doing it now.

File: src/class/SimpleData.ts [hunk 3]
```diff
@@ -125,45 +71,62 @@ export default class SimpleData {
...
-    getData(): SimpleDataItem[] {
-        return this._data
-    }
+    // *** IMPORTING METHOD *** //
 
     @logCall()
-    getKeys(): string[] {
-        return this._keys
-    }
+    loadDataFromArray({
```
[the remark this answers is not in the document]
Comment (nshiab) [reply src/class/SimpleData.ts:77]: Ok! Doing it. I wasn't a fan of the loadDataFromArray.

File: src/class/SimpleData.ts [hunk 4]
```diff
@@ -463,83 +487,56 @@ export default class SimpleData {
...
 
-    @logCall()
-    excludeOutliers({ key, overwrite = true }: { key: string, overwrite?: boolean }): SimpleData {
-        const data = excludeOutliers_(this._data, key, this.verbose)
-        overwrite && this.#updateSimpleData(data)
+    // *** VISUALIZATION METHODS *** //
 
-        return this
+    @logCall()
+    createChart({ type, x, y, color }: { type: "dot" | "line" | "bar" | "box", x: string, y: string, color?: string }): string {
```
[the remark this answers is not in the document]
Comment (nshiab) [reply src/class/SimpleData.ts:493]: Ok!

File: src/class/SimpleData.ts [hunk 5]
```diff
@@ -463,83 +487,56 @@ export default class SimpleData {
...
-    }): SimpleData {
-        const data = addItems_(
-            this._data,
-            dataToBeAdded,
-            this.verbose
-        )
-        overwrite && this.#updateSimpleData(data)
-
-        return this
+    createCustomChart({ plotOptions }: { plotOptions: object }): string {
```
[the remark this answers is not in the document]
Comment (nshiab) [reply src/class/SimpleData.ts:499]: Done

File: src/class/SimpleDataNode.ts [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+    }
+
+    @logCall()
+    saveChart({ path, type, x, y, color }: { path: string, type: "dot" | "line" | "bar" | "box", x: string, y: string, color?: string }): string {
+        if (global.document === undefined) {
+            const jsdom = new JSDOM("")
+            global.document = jsdom.window.document
+        }
+        const chart = super.createChart({ x, y, type, color })
+        writeChart(path, chart, this.verbose)
```
[the remark this answers is not in the document]
Comment (nshiab) [reply src/class/SimpleDataNode.ts:71]: Ok!

**Review by isabellebouchard — APPROVED** [review]
👍

**Pull request merged**